server/routes.go

```diff
@@ -294,19 +292,23 @@ func GenerateHandler(c *gin.Context) {
 				resp.LoadDuration = checkpointLoaded.Sub(checkpointStart)
 
 				if !req.Raw {
-					// append the generated text to the history and template it if needed
-					promptVars.Response = generated.String()
-					result, err := model.PostResponseTemplate(promptVars)
+					p, err := Prompt(model.Template, req.System, req.Prompt, generated.String(), false)
```

Comment:
```suggestion
					p, err := Prompt(req.Template, req.System, req.Prompt, generated.String(), false)
```

If the comment about `req.Template` being the source of truth is correct